test/testdata/lsp/fast_path/ivar_redeclare.2.rbupdate

```diff
@@ -1,5 +1,5 @@
 # typed: true
-# assert-slow-path: true
+# assert-fast-path: ivar_redeclare.rb
```

Comment:
One more suggestion for a test:

Do you want to add one that changes the a `prop` or `const` in a `T::Struct`?

Doing so should change the getter method, setter method, constructor, and ivar types, but we should be able to take the fast path now.

Reply:
(Possibly larger question: might be worth having some tests that intermix renaming methods and renaming ivars? Maybe rename the method that declares the ivar?)

Reply:
I added some basic tests for each of these.  It's really cool to see editing a `prop` take the fast path!